libs/common/src/sql-db-connection/migrations/1717746522387-PaymentTableUpdateAmountColumns.ts

```diff
@@ -0,0 +1,69 @@
+import { MigrationInterface, QueryRunner } from 'typeorm';
+
+/**
+ * This migration has been manually modified in order to work
+ * with ledger tables.
+ */
+export class PaymentTableUpdateAmountColumns1717746522387 implements MigrationInterface {
+  name = 'PaymentTableUpdateAmountColumns1717746522387';
+
+  public async up(queryRunner: QueryRunner): Promise<void> {
+    // Create the new columns with the required data-type and set to nullable
+    await queryRunner.query(`
+            ALTER TABLE "Payment"
+            ADD "amount" decimal(14, 2) NULL
+        `);
+    await queryRunner.query(`
+            ALTER TABLE "Payment"
+            ADD "reference" nvarchar(255) NULL
+        `);
+
+    // Copy all the values from the old columns to the new columns
+    await queryRunner.query(`
+            UPDATE "Payment" SET "amount" = "amountReceived"
+        `);
+    await queryRunner.query(`
+            UPDATE "Payment" SET "reference" = "paymentReference"
+        `);
+
+    // Delete the old columns
+    await queryRunner.query(`
+            ALTER TABLE "Payment"
+            DROP COLUMN "amountReceived"
+        `);
+    await queryRunner.query(`
+            ALTER TABLE "Payment"
+            DROP COLUMN "paymentReference"
+        `);
+  }
+
+  public async down(queryRunner: QueryRunner): Promise<void> {
+    // Create the new columns with the required data-type and set to nullable
+    await queryRunner.query(`
+            ALTER TABLE "Payment"
+            ADD "amountReceived" int NULL
+        `);
+    await queryRunner.query(`
+            ALTER TABLE "Payment"
+            ADD "paymentReference" nvarchar(255) NULL
+        `);
+
+    // Copy all the values from the old columns to the new columns
+    await queryRunner.query(`
+            UPDATE "Payment" SET "amountReceived" = "amount"
```

Comment:
This technically might not work going from a decimal to a non decimal but I think leaving this in is fine - we don't expect to run the down migrations much on a non-empty database that has had new data put into it/we could manually get around this. Just noting this for anyone looking at the PR in the future!

Reply:
I've confirmed that the migration does work in the other direction even with entries with decimal amounts in the table - the amount is just converted to an integer